src/modules/MouseUtils/MouseHighlighter/dllmain.cpp

```diff
@@ -0,0 +1,316 @@
+        {
+            // Improper JSON.
+        }
+    }
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        m_enabled = true;
+        Trace::EnableMouseHighlighter(true);
+        std::thread([=]() { MouseHighlighterMain(m_hModule, m_highlightSettings); }).detach();
+    }
+
+    // Disable the powertoy
+    virtual void disable()
+    {
+        m_enabled = false;
+        Trace::EnableMouseHighlighter(false);
+        MouseHighlighterDisable();
+    }
+
+    // Returns if the powertoys is enabled
+    virtual bool is_enabled() override
+    {
+        return m_enabled;
+    }
+
+    virtual std::optional<HotkeyEx> GetHotkeyEx() override
+    {
+        return m_hotkey;
+    }
+
+    virtual void OnHotkeyEx() override
+    {
+        MouseHighlighterSwitch();
+    }
+
+    // Load the settings file.
+    void init_settings()
+    {
+        try
+        {
+            // Load and parse the settings file for this PowerToy.
+            PowerToysSettings::PowerToyValues settings =
+                PowerToysSettings::PowerToyValues::load_from_settings_file(MouseHighlighter::get_key());
+            parse_settings(settings);
+        }
+        catch (std::exception&)
+        {
+            // Error while loading from the settings file. Let default values stay as they are.
+        }
+    }
+
+    void parse_settings(PowerToysSettings::PowerToyValues& settings)
+    {
+        auto settingsObject = settings.get_raw_json();
+        MouseHighlighterSettings highlightSettings;
+        if (settingsObject.GetView().Size())
+        {
+            try
+            {
+                // Parse HotKey
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_ACTIVATION_SHORTCUT);
+                auto hotkey = PowerToysSettings::HotkeyObject::from_json(jsonPropertiesObject);
+                m_hotkey = HotkeyEx();
+                if (hotkey.win_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_WIN;
+                }
+
+                if (hotkey.ctrl_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_CONTROL;
+                }
+
+                if (hotkey.shift_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_SHIFT;
+                }
+
+                if (hotkey.alt_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_ALT;
+                }
+
+                m_hotkey.vkCode = hotkey.get_code();
+            }
+            catch (...)
+            {
+                Logger::warn("Failed to initialize Mouse Highlighter activation shortcut");
+            }
+            uint8_t opacity = MOUSE_HIGHLIGHTER_DEFAULT_OPACITY;
+            try
+            {
+                // Parse Opacity
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_HIGHLIGHT_OPACITY);
```

Comment:
couldn't we use `src\common\utils\json.h` in this file?

Reply:
I've based the code on what I saw for ShortcutGuide and some other PowerToys.
So we have two different libs for this?
What's the issue with using this one vs the other?